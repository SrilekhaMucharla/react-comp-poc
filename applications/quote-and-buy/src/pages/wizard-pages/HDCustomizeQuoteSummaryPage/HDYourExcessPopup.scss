//TODO: check font family
.popup-your-excess {
    color: $app-theme;
}

.popup-your-excess__description {
    margin-top: 5%;
    margin-bottom: 5%;
    font-size: $font-size-md;
    line-height: 24px;
    color: $info-color;
}
.popup-your-excess__tables {
    table {
        margin-top: 1.25em;
    }

    table:first-child {
        margin-top: 0;
    }
}

.popup-your-excess__windowscreen {
    .excess-table-ref__row__th.excess-table-ref__col-1 {
        width: 20%;
    }
}

.excess-info-card {
    margin-bottom: 5%;
}
