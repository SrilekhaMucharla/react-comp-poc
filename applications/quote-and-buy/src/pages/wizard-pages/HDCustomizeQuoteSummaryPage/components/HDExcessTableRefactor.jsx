import PropTypes from 'prop-types';
import React from 'react';
import * as messages from '../HDExcessTable.messages';

const HDExcessTableRefactor = ({
    drivers,
    excess,
    hideTotalColumn,
    // eslint-disable-next-line react/prop-types
    tableColumnOne,
    // eslint-disable-next-line react/prop-types
    tableColumnTwo,
    className
}) => {
    const displayAmount = (value) => `Â£${value}`;
    const { excessName, compulsoryAmount, voluntaryAmount } = excess;

    return (
        <table className={`excess-table-ref ${className}`}>
            <thead className="excess-table-ref__head">
                <tr className="excess-table-ref__row">
                    <th className="excess-table-ref__row__th excess-table-ref__name">{excessName}</th>
                    <th className="excess-table-ref__row__th excess-table-ref__col-1">{tableColumnOne}</th>
                    <th className="excess-table-ref__row__th excess-table-ref__col-2">{tableColumnTwo}</th>
                    {(hideTotalColumn) ? <th /> : <th className="excess-table-ref__row__th excess-table-ref__total-header">{messages.totalHeader}</th>}
                </tr>
            </thead>
            <tbody className="excess-table-ref__body">
                {drivers.map((driver) => (
                    <tr className="excess-table-ref__body-row" key={driver.id ? driver.id : ''}>
                        <td className="excess-table-ref__body-row__td excess-table-ref__excess-name">{driver.name}</td>
                        <td className="excess-table-ref__body-row__td excess-table-ref__compulsory-amount">{driver.excesses ? displayAmount(driver.excesses.compulsoryAmount) : displayAmount(compulsoryAmount)}</td>
                        <td className="excess-table-ref__body-row__td excess-table-ref__voluntary-amount">{displayAmount(voluntaryAmount)}</td>
                        {(hideTotalColumn)
                            ? <td className="excess-table-ref__body-row__td" />
                            : <td className="excess-table-ref__body-row__td excess-table-ref__total-amount">{driver.excesses ? displayAmount(driver.excesses.compulsoryAmount + voluntaryAmount) : displayAmount(voluntaryAmount + compulsoryAmount)}</td> }
                    </tr>
                ))}
            </tbody>
        </table>
    );
};

export const ExcessProps = PropTypes.shape({
    excessName: PropTypes.string.isRequired,
    voluntaryAmount: PropTypes.number.isRequired,
    compulsoryAmount: PropTypes.number.isRequired,
});

HDExcessTableRefactor.propTypes = {
    excess: PropTypes.arrayOf(ExcessProps).isRequired,
    // eslint-disable-next-line react/require-default-props
    hideTotalColumn: PropTypes.bool,
    className: PropTypes.string,
    drivers: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string,
        id: PropTypes.number
    })).isRequired
};

HDExcessTableRefactor.defaultProps = {
    className: '',
};

export default HDExcessTableRefactor;
