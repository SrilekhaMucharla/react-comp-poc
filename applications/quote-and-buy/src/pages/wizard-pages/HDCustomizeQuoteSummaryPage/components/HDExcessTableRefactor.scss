.excess-table-ref {
    width: 100%;
    margin-bottom: 20px;
    font-size: $font-size-md;
}

.excess-table-ref__head {
    color: $app-theme;
    background-color: $blue-transparent-10;
    border-bottom: 1px solid $blue-light;
    @include media-breakpoint-down(sm) {
        font-size: $font-size-sm;
    }
}

.excess-table-ref__body {
    background-color: $gray-light-7;
    color: $info-color;
    @include media-breakpoint-down(sm) {
        font-size: $font-size-sm;
    }

    tr:nth-child(even) {
        background-color: #f7f7f9;
    }
}

.excess-table-ref__body-row {
    height: 40px;
}

.excess-table-ref__body-row__td {
    text-align: right;
    width: 15%;
    padding: 5px;

    &:last-child {
        padding-right: 13px;
        font-weight: 500;
    }
}

.excess-table-ref__row {
    height: 54px;
}

.excess-table-ref__row__th {
    font-weight: 500;
    text-align: right;
    width: 15%;
    padding: 5px;

    &:last-child {
        padding-right: 13px;
        font-weight: bold;
    }
}

.excess-table-ref__name {
    padding-left: 13px;
    font-weight: bold;
    text-align: left;
    width: 30%;
}

.excess-table-ref__excess-name {
    padding-left: 13px;
    font-weight: 500;
    text-align: left;
    width: 40%;
}

.excess-table-ref__row__th.excess-table-ref__col-2 {
    width: 25%;
}