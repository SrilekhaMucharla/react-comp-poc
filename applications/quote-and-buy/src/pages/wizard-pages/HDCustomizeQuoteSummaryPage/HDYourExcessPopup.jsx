import { Col, Row } from 'react-bootstrap';
import { HDLabelRefactor, HDInfoCardRefactor } from 'hastings-components';
import _ from 'lodash';
import PropTypes from 'prop-types';
import React from 'react';
import HDExcessTableRefactor, { ExcessProps } from './components/HDExcessTableRefactor';
import * as messages from './HDYourExcessPopup.messages';
import InfoIcon from './svg/InfoIcon';
import tipCirclePurple from '../../../assets/images/icons/tip_circle_purple.svg';
import {
    AnalyticsHDOverlayPopup as HDOverlayPopup
} from '../../../web-analytics';
import useNareApplicable from '../__helpers__/nareHelper';

const HDYourExcessPopup = ({
    drivers, globalExcesses, customizeSubmissionVM, pcCurrentDate, pageMetadata, className
}) => {
    const allExcesses = [];
    const branchCode = _.get(customizeSubmissionVM, 'value.quote.branchCode', '');
    const periodStartDate = _.get(customizeSubmissionVM, 'value.periodStartDate', '');
    const excessInfoMessage = (branchCode === messages.HE) ? messages.excessMessageHE : messages.excessMessage;

    drivers.map(({ excesses }) => {
        if (allExcesses.find((el) => (el.excessName === excesses.excessName)) === undefined) return allExcesses.push(excesses);
        return null;
    });
    return (
        <HDOverlayPopup
            webAnalyticsEvent={{ event_action: `${messages.customizeQuote} - ${messages.mainHeader}` }}
            webAnalyticsView={{ ...pageMetadata, page_section: `${messages.customizeQuote} - ${messages.mainHeader}` }}
            id="popup-your-excess"
            className={`popup-your-excess-container${(className) ? ` ${className}` : ''}`}
            overlayButtonIcon={<InfoIcon width="20" height="20" color="#0085ff" />}
        >
            <Row className="popup-your-excess">
                <Col>
                    <Row>
                        <Col>
                            <HDLabelRefactor
                                Tag="h2"
                                text={messages.mainHeader}
                                size="lg" />
                        </Col>
                    </Row>
                    <Row className="popup-your-excess__tables">
                        <Col>
                            <Row>
                                <Col>
                                    {(drivers && drivers.length > 0 && allExcesses && allExcesses.length > 0) && allExcesses.map((excess) => (
                                        <HDExcessTableRefactor
                                            excess={excess}
                                            drivers={drivers}
                                            tableColumnOne={messages.Compulsory}
                                            tableColumnTwo={messages.Voluntary} />
                                    ))}
                                </Col>
                            </Row>
                            <Row>
                                <Col>
                                    {(globalExcesses && globalExcesses.length > 0) && globalExcesses.map((excess) => (
                                        <HDExcessTableRefactor
                                            className={(excess.excessName === messages.windowGlassText) ? 'popup-your-excess__windowscreen' : ''}
                                            drivers={[{ id: 1, name: messages.forAllDriversHeader }]}
                                            tableColumnOne={(excess.excessName === messages.windowGlassText) ? messages.Repair : messages.Compulsory}
                                            tableColumnTwo={(excess.excessName === messages.windowGlassText) ? messages.Replacement : messages.Voluntary}
                                            hideTotalColumn={(excess.excessName === messages.windowGlassText)}
                                            excess={excess} />
                                    ))}
                                </Col>
                            </Row>
                        </Col>
                    </Row>
                    <Row>
                        {(periodStartDate && pcCurrentDate && useNareApplicable(periodStartDate, pcCurrentDate)
                        && (
                            <HDInfoCardRefactor
                                id="excess-info-card"
                                image={tipCirclePurple}
                                paragraphs={[excessInfoMessage]}
                                theme="light"
                                size="thin"
                                className="mt-3 mt-md-4 excess-info-card" />
                        ))}
                    </Row>
                    <Row>
                        <Col>
                            <HDLabelRefactor
                                Tag="h3"
                                text={messages.overlayHeaderOne} />
                        </Col>
                    </Row>
                    <Row>
                        <Col>
                            <HDLabelRefactor
                                className="popup-your-excess__description"
                                text={messages.overlayBodyOne}
                                Tag="p" />
                            <HDLabelRefactor
                                className="popup-your-excess__description"
                                text={messages.overlayBodyTwo}
                                Tag="p" />
                            <HDLabelRefactor
                                className="popup-your-excess__description"
                                text={messages.overlayBodyThree}
                                Tag="p" />
                        </Col>
                    </Row>
                    <Row>
                        <Col>
                            <HDLabelRefactor
                                Tag="h3"
                                text={messages.overlayHeaderTwo} />
                        </Col>
                    </Row>
                    <Row>
                        <Col>
                            <HDLabelRefactor
                                className="popup-your-excess__description"
                                text={messages.overlayBodyVoluntaryExcess}
                                Tag="p" />
                        </Col>
                    </Row>
                    <Row>
                        <Col>
                            <HDLabelRefactor
                                Tag="h3"
                                text={messages.overlayHeaderThree} />
                        </Col>
                    </Row>
                    <Row>
                        <Col>
                            <HDLabelRefactor
                                className="popup-your-excess__description"
                                text={messages.overBodyCompulsoryExcess}
                                Tag="p" />
                        </Col>
                    </Row>
                    <Row>
                        <Col>
                            <HDLabelRefactor
                                Tag="h3"
                                text={messages.overlayHeaderFour} />
                        </Col>
                    </Row>
                    <Row>
                        <Col>
                            <HDLabelRefactor
                                className="popup-your-excess__description"
                                text={messages.overBodyDesc}
                                Tag="p" />
                        </Col>
                    </Row>
                </Col>
            </Row>
        </HDOverlayPopup>
    );
};

HDYourExcessPopup.defaultProps = {
    className: null,
    customizeSubmissionVM: null
};

HDYourExcessPopup.propTypes = {
    drivers: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string.isRequired,
        excesses: PropTypes.arrayOf(ExcessProps).isRequired
    })).isRequired,
    globalExcesses: PropTypes.arrayOf(ExcessProps).isRequired,
    customizeSubmissionVM: PropTypes.shape({ value: PropTypes.object }),
    pageMetadata: PropTypes.shape({
        page_name: PropTypes.string.isRequired,
        page_type: PropTypes.string.isRequired,
        sales_journey_type: PropTypes.string.isRequired
    }).isRequired,
    className: PropTypes.string
};

export default HDYourExcessPopup;
