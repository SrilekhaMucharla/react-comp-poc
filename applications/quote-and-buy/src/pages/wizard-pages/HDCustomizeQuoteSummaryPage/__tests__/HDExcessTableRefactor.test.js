import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import HDExcessTableRefactor from '../components/HDExcessTableRefactor';
import * as messages from '../HDYourExcessPopup.messages';

Enzyme.configure({ adapter: new Adapter() });

describe('<HDExcessTableRefactor />', () => {
    test('render with mandatory props', () => {
        const wrapper = shallow((
            <HDExcessTableRefactor
                name="James Bond"
                drivers={[{ id: 1, name: messages.forAllDriversHeader }]}
                excess={[
                    { excessName: messages.windowGlassText, voluntaryAmount: 200, compulsoryAmount: 300 },
                ]} />
        ));
        expect(wrapper).toMatchSnapshot();
    });
    test('render with mandatory props and hide total column', () => {
        const wrapper = shallow((
            <HDExcessTableRefactor
                name="James Bond"
                drivers={[{ id: 1, name: messages.forAllDriversHeader }]}
                hideTotalColumn
                excess={[
                    { excessName: messages.windowGlassText, voluntaryAmount: 200, compulsoryAmount: 300 },
                ]} />
        ));
        expect(wrapper).toMatchSnapshot();
    });
});
