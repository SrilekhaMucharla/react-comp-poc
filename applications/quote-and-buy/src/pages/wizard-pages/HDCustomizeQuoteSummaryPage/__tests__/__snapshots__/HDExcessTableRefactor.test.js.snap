// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDExcessTableRefactor /> render with mandatory props 1`] = `
<table
  className="excess-table-ref "
>
  <thead
    className="excess-table-ref__head"
  >
    <tr
      className="excess-table-ref__row"
    >
      <th
        className="excess-table-ref__row__th excess-table-ref__name"
      />
      <th
        className="excess-table-ref__row__th excess-table-ref__col-1"
      />
      <th
        className="excess-table-ref__row__th excess-table-ref__col-2"
      />
      <th
        className="excess-table-ref__row__th excess-table-ref__total-header"
      >
        Total
      </th>
    </tr>
  </thead>
  <tbody
    className="excess-table-ref__body"
  >
    <tr
      className="excess-table-ref__body-row"
      key="1"
    >
      <td
        className="excess-table-ref__body-row__td excess-table-ref__excess-name"
      >
        All drivers
      </td>
      <td
        className="excess-table-ref__body-row__td excess-table-ref__compulsory-amount"
      >
        £undefined
      </td>
      <td
        className="excess-table-ref__body-row__td excess-table-ref__voluntary-amount"
      >
        £undefined
      </td>
      <td
        className="excess-table-ref__body-row__td excess-table-ref__total-amount"
      >
        £NaN
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<HDExcessTableRefactor /> render with mandatory props and hide total column 1`] = `
<table
  className="excess-table-ref "
>
  <thead
    className="excess-table-ref__head"
  >
    <tr
      className="excess-table-ref__row"
    >
      <th
        className="excess-table-ref__row__th excess-table-ref__name"
      />
      <th
        className="excess-table-ref__row__th excess-table-ref__col-1"
      />
      <th
        className="excess-table-ref__row__th excess-table-ref__col-2"
      />
      <th />
    </tr>
  </thead>
  <tbody
    className="excess-table-ref__body"
  >
    <tr
      className="excess-table-ref__body-row"
      key="1"
    >
      <td
        className="excess-table-ref__body-row__td excess-table-ref__excess-name"
      >
        All drivers
      </td>
      <td
        className="excess-table-ref__body-row__td excess-table-ref__compulsory-amount"
      >
        £undefined
      </td>
      <td
        className="excess-table-ref__body-row__td excess-table-ref__voluntary-amount"
      >
        £undefined
      </td>
      <td
        className="excess-table-ref__body-row__td"
      />
    </tr>
  </tbody>
</table>
`;
